File: haskell/src/Solutions/Euler105.hs
module Solutions.Euler105 where
import Utils.Input(getInput)
import Solutions.Euler103(testSet)

read' :: String -> [Int]
read' s = read $ "[" ++ s ++ "]"

euler105 :: IO String
euler105 = show . sum . map sum . filter testSet . map read' . lines <$> getInput 105

File: rust/src/solutions/euler105.rs
use crate::solutions::euler103::test_set;
use crate::utils::input::get_input;

pub fn euler105() -> String {
    get_input(105)
        .lines()
        .map(|line| {
            line.split(',')
                .map(|n| n.parse::<u32>().unwrap())
                .collect::<Vec<u32>>()
        })
        .flat_map(|vec| {
            if test_set(&vec) {
                Some(vec.into_iter().sum::<u32>())
            } else {
                None
            }
        })
        .sum::<u32>()
        .to_string()
}
